File: Div3-Group1/MaximumAverageSubarrayI.cpp
class Solution {
public:
    double findMaxAverage(vector<int>& nums, int k) {
        double max=0;
        double average=0;
        int kDouble=k;

        while(kDouble--)
        {
            max+=nums[kDouble];
        }
        
        average=max;
        max/=k;

        for(int i=k; i<nums.size();i++)
        {
            average+=nums[i];
            average-=nums[i-k];

            if(max<(average/k))
                max=average/k;
        }

        return max;
    }
};

File: leet-code-solutions/maximum-average-subarray-i.py
class Solution:
    def findMaxAverage(self, nums: List[int], k: int) -> float:
        max_=float('-inf')
        left=0
        right=k-1
        sum_=sum(nums[left:right])

        print(nums[left:right])

        while left < len(nums)-k+1:
            sum_ += nums[right]
            
            avg = sum_/k
            max_=max(max_,avg)
        
            sum_ -= nums[left]

            left += 1
            right += 1
            # print(nums[left:right])

        return max_